File: BeakJoon/Silver5/11723/11723.cpp
#include <stdio.h>
#include <string.h>

int main()
{
    int n, commandNumber;
    int set[21] = {
        0,
    };
    char command[7] = {};
    scanf("%d", &n);
    for (int i = 0; i < n; i++)
    {
        scanf("%s", command);
        if (!strcmp(command, "add"))
        {
            scanf("%d", &commandNumber);
            set[commandNumber] = 1;
        }
        else if (!strcmp(command, "remove"))
        {
            scanf("%d", &commandNumber);
            set[commandNumber] = 0;
        }
        else if (!strcmp(command, "toggle"))
        {
            scanf("%d", &commandNumber);
            set[commandNumber] = !set[commandNumber];
        }
        else if (!strcmp(command, "all"))
        {
            for (int j = 1; j <= 20; j++)
                set[j] = 1;
        }
        else if (!strcmp(command, "empty"))
        {
            for (int j = 1; j <= 20; j++)
                set[j] = 0;
        }
        else
        {
            scanf("%d", &commandNumber);
            printf("%d\n", set[commandNumber]);
        }
    }
}

File: BeakJoon/Silver5/11723/11723.js
"use strict";
//정답은 맞는 것 같은데 백준 문제로 메모리 초과


const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().split("\n");

const M = input.shift();
input = input.map((v) => v.replace(/\r/g, ""));
let i, j;
let set = new Set();

for (i = 0; i < M; i++) {
  const [cmd, num] = input.shift().split(" ");
  if (cmd === "add") {
    set.add(num);
  } else if (cmd === "remove") {
    set.delete(num);
  } else if (cmd === "check") {
    console.log(set.has(num) ? 1 : 0);
  } else if (cmd === "toggle") {
    if (set.has(num)) set.delete(num);
    else set.add(num);
  } else if (cmd === "all") {
    for (j = 1; j < 21; j++) set.add(j);
  } else {
    set.clear();
  }
}
